app/services/discourse_ai/tool_name_checker.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+module DiscourseAi
+  class ToolNameChecker
+    def initialize(tool_name)
+      @tool_name = tool_name
+    end
+
+    def check
+      if @tool_name.match? AiTool::ALPHANUMERIC_PATTERN
+        check_name_availability
+      else
+        { available: false, errors: [I18n.t("discourse_ai.tools.name.characters")] }
+      end
+    end
+
+    private
+
+    def check_name_availability
```

Comment:
Should we also add a unique index to ensure uniqueness ?

Reply:
Thanks, I'll double-check it later, but we did have that unique option on that column in the first place https://github.com/discourse/discourse-ai/blob/main/db/migrate/20240618080148_create_ai_tools.rb#L6, it's highly likely that we used the wrong option syntax for it, it should be

```ruby
t.string :name, index: { unique: true }
```